spam/02_spam_data_augmentation_tutorial.py

```diff
@@ -291,59 +309,51 @@ def replace_adjective_with_synonym(x):
 sess = tf.compat.v1.Session(graph=tf.compat.v1.get_default_graph(), config=session_conf)
 tf.compat.v1.keras.backend.set_session(sess)
 
-
 # %% [markdown]
-# Next, we add some boilerplate code for creating an LSTM model.
-
-# %%
-def get_lstm_model(num_buckets, embed_dim=16, rnn_state_size=64):
-    lstm_model = tf.keras.Sequential()
-    lstm_model.add(tf.keras.layers.Embedding(num_buckets, embed_dim))
-    lstm_model.add(tf.keras.layers.LSTM(rnn_state_size, activation=tf.nn.relu))
-    lstm_model.add(tf.keras.layers.Dense(1, activation=tf.nn.sigmoid))
-    lstm_model.compile("Adagrad", "binary_crossentropy", metrics=["accuracy"])
-    return lstm_model
-
-
-# %%
-def train_and_test(train_set, Y_train, num_buckets=30000):
-    def map_pad_or_truncate(string, max_length=30):
-        """Tokenize text, pad or truncate to get max_length, and hash tokens."""
-        ids = tf.keras.preprocessing.text.hashing_trick(
-            string, n=num_buckets, hash_function="md5"
-        )
-        return ids[:max_length] + [0] * (max_length - len(ids))
-
-    # Tokenize training text and convert words to integers.
-    X_train = np.array(list(map(map_pad_or_truncate, train_set.text)))
-    lstm_model = get_lstm_model(num_buckets)
-
-    # Tokenize validation set text and convert words to integers.
-    X_valid = np.array(list(map(map_pad_or_truncate, df_valid.text)))
-
+# Now we'll train our LSTM on both the original and augmented data sets to compare performance.
+
+# %% {"tags": ["md-exclude-output"]}
+from utils import featurize_df_tokens, get_keras_lstm, get_keras_early_stopping
+
+X_train = featurize_df_tokens(df_train)
+X_train_augmented = featurize_df_tokens(df_train_augmented)
+X_valid = featurize_df_tokens(df_valid)
+X_test = featurize_df_tokens(df_test)
+
+
+def train_and_test(
+    X_train,
+    Y_train,
+    X_valid=X_valid,
+    Y_valid=Y_valid,
+    X_test=X_test,
+    Y_test=Y_test,
+    num_buckets=30000,
+):
+    lstm_model = get_keras_lstm(num_buckets)
```

Comment:
I think this is great but should include a comment saying something to the effect of 'this is *just* a vanilla LSTM in Keras that we put in the util file for cleanliness'